lib/jnpr/junos/utils/fs.py

```diff
@@ -268,30 +270,34 @@ def _decode(fs):
     # directory_usage - filesystem directory usage
     # -------------------------------------------------------------------------
 
-    def directory_usage(self, path="."):
+    def directory_usage(self, path=".", depth=0):
         """
         Returns the directory usage, similar to the unix "du" command.
 
-        :returns: approximate directory usage, including subdirectories, in bytes
+        :returns: dict of directory usage, including subdirectories if depth > 0
         """
-        rsp = self._dev.rpc.get_directory_usage_information(path=path, depth="0")
+        BLOCK_SIZE = 512
 
-        used_space = rsp.findtext("directory/used-space")
+        rsp = self._dev.rpc.get_directory_usage_information(path=path, depth=str(depth))
 
-        if used_space is None:
-            return None
+        if rsp.findtext("directory/used-space") is None:
+            # Likely, no such directory
+            raise RpcError(rsp=rsp)
```

Comment:
I'm sorry. I know I suggested this change, but now I'm rethinking it.
This would only raise an `RpcError` if there was no `<directory>` element in the response that contained a `<used-space>` element. If there were multiple `<directory>` elements in the response, but only some of them were missing the `<used-space>` element, it wouldn't address that situation.

Now that you've modified to return a dictionary, we probably need to handle the exceptional conditions of missing elements within the loop over each `<directory>` element. I suggest removing lines 283-285.